### PR TITLE
clearer unknown contract call formatting

### PR DESCRIPTION
**Pull request by rainbreak**
@MrChico

### CODE REVIEW

File: src/hevm/src/EVM/Format.hs [hunk 1]
```diff
@@ -138,7 +149,7 @@ formatSBinary (ConcreteBuffer bs) = formatBinary bs
 showTraceTree :: DappInfo -> VM -> Text
 showTraceTree dapp =
   traceForest
-    >>> fmap (fmap (unpack . showTrace dapp))
+    >>> fmap (fmap (unpack . (showTrace dapp)))
```
Comment (MrChico) [inline src/hevm/src/EVM/Format.hs:152]:
```suggestion
    >>> fmap (fmap (unpack . showTrace dapp))
```

File: src/hevm/src/EVM/Format.hs [hunk 2]
```diff
@@ -213,20 +235,32 @@ showTrace dapp trace =
       t
     FrameTrace (CreationContext hash _ _ ) ->
       "create " <> maybeContractName (preview (dappSolcByHash . ix hash . _2) dapp) <> pos
-    FrameTrace (CallContext _ _ hash abi calldata _ _) ->
-      case preview (dappSolcByHash . ix hash . _2) dapp of
+    FrameTrace (CallContext target context _ _ hash abi calldata _ _) ->
+      let calltype = if (target == context)
+                     then "call "
+                     else "delegatecall "
```
Comment (MrChico) [inline src/hevm/src/EVM/Format.hs:241]: this could also be `CALLCODE` I believe

File: src/hevm/src/EVM/Format.hs [hunk 3]
```diff
@@ -213,20 +235,32 @@ showTrace dapp trace =
       t
     FrameTrace (CreationContext hash _ _ ) ->
       "create " <> maybeContractName (preview (dappSolcByHash . ix hash . _2) dapp) <> pos
-    FrameTrace (CallContext _ _ hash abi calldata _ _) ->
-      case preview (dappSolcByHash . ix hash . _2) dapp of
+    FrameTrace (CallContext target context _ _ hash abi calldata _ _) ->
+      let calltype = if (target == context)
```
Comment (MrChico) [inline src/hevm/src/EVM/Format.hs:239]:
```suggestion
      let calltype = if target == context
```

File: src/hevm/src/EVM/Format.hs (hunk 2 above)
Comment (rainbreak) [reply src/hevm/src/EVM/Format.hs:241]: Not as simple to determine unfortunately, so it will just show up as delegatecall.

Comment (MrChico) [reply src/hevm/src/EVM/Format.hs:241]: should we display it as `delegatecall/callcode`?

Comment (rainbreak) [reply src/hevm/src/EVM/Format.hs:241]: Imo no because it's clutter the vast majority of the time. Does anyone use callcode anymore?

Comment (rainbreak) [reply src/hevm/src/EVM/Format.hs:241]: If we really wanted to show it we could add a new `calltype` field to `CallContext`.

Comment (rainbreak) [reply src/hevm/src/EVM/Format.hs:241]: If you still think it's worth it, then how about also distinguishing `create` and `create2`?

Comment (MrChico) [reply src/hevm/src/EVM/Format.hs:241]: ok, I'm fine with `delegatecall` for now

**Review by MrChico — APPROVED** [review]

File: src/hevm/src/EVM/Format.hs (hunk 2 above)
Comment (rainbreak) [reply src/hevm/src/EVM/Format.hs:241]: We also don't distinguish `staticcall`.

**Pull request merged**